lib/ts/recipe/emailpassword/recipe.ts

```diff
@@ -101,6 +105,183 @@ export default class Recipe extends RecipeModule {
+                                // - tenant1 -> emailpassword with email e1
+                                // - tenant2 -> thirdparty with email e2 (e2 can be equal to e1, doesn't matter)
+                                // now if the user has logged into tenant2, and is adding
+                                // a password to that, we do return ["emailpassword"]
+                                // cause it's there in tenant1 (i.e. we do not check
+                                // for tenant below), then the frontend will end up
+                                // calling the signIn API instead of signUp, and since
+                                // these APIs are tenant specific, then it will return
+                                // wrong credentials error even if the password is correct.
+
+                                // Notice that we also check for if the email is fake or not,
+                                // cause if it is fake, then we should not consider it as setup
+                                // so that the frontend asks the user to enter an email,
+                                // or uses the email of another login method.
+                                if (
+                                    loginMethod.recipeId === Recipe.RECIPE_ID &&
+                                    loginMethod.tenantIds.includes(tenantId) &&
+                                    !isFakeEmail(loginMethod.email!)
+                                ) {
+                                    return ["emailpassword"];
+                                }
+                            }
+                            return [];
+                        });
+
+                        mfaInstance.addGetEmailsForFactorFromOtherRecipes((user: User, sessionRecipeUserId) => {
+                            // This function is called in the MFA info endpoint API.
+                            // Based on https://github.com/supertokens/supertokens-node/pull/741#discussion_r1432749346
+
+                            // preparing some reusable variables for the logic below...
+                            let sessionLoginMethod = user.loginMethods.find((lM) => {
+                                return lM.recipeUserId.getAsString() === sessionRecipeUserId.getAsString();
+                            });
+                            if (sessionLoginMethod === undefined) {
+                                // this can happen maybe cause this login method
+                                // was unlinked from the user or deleted entirely...
+                                return {
+                                    status: "UNKNOWN_SESSION_RECIPE_USER_ID",
+                                };
+                            }
+
+                            const orderedLoginMethodsByTimeJoinedOldestFirst = user.loginMethods.sort((a, b) => {
+                                return a.timeJoined - b.timeJoined;
+                            });
```

Comment:
does this sort correctly such that oldest time joined is 0th index?

Reply:
yes it does, least timeJoined means the oldest